src/functional/otf/languages.py

```diff
@@ -0,0 +1,63 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+from __future__ import annotations
+
+import dataclasses
+from typing import ClassVar, Optional
+
+
+class LanguageTag:
+    """
+    Represent a programming language.
+
+    ``.settings_level`` should be set to the ``LanguageSettings`` subclass
+    with the minimum amount of settings required for the language.
+    """
+
+    settings_level: ClassVar[type[LanguageSettings]]
```

Comment:
What does *level* stand for? The name is not clear to me

Reply:
The mental model was level of detail. The LanguageSettings are currently implemented as a hierarchy with the most general one at the top and successively more detailed ones at the bottom. Each language has a minimum level of detail it needs.

As always, open for better suggestions if (better = supported by a more applicable mental model).

Reply:
I see. It's okay for me then. What's your opinion about `settings_class` or `settings_type`, which makes it clear that you expect a type (though it's in the annotation) and also allows for having a non-linear structure for language settings?

Reply:
Fine with me.